### PR TITLE
add net_trace.stp

### PR DESCRIPTION
**Pull request by LinuxGit**
saddr and daddr's order in  `tcp.sendmsg.return` is ambiguous,  need to be verified.

### CODE REVIEW

File: systemtap/tools/net_trace.stp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+probe kernel.function("tcp_accept").return?,
+      kernel.function("inet_csk_accept").return? {
+    if (target() != pid()) next
+
+    sock = $return
+    if (sock != 0)
+        printf("%15s[%ld] - port %d accept request from %s \n", execname(), tid(), inet_get_local_port(sock), inet_get_ip_source(sock))
+}
+
+// probe tcp.receive {
```
Comment (siddontang) [inline systemtap/tools/net_trace.stp]: remove this

File: systemtap/tools/net_trace.stp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+probe tcp.sendmsg.return
+{
+    if (target() != pid()) next
+
+    saddr = format_ipaddr(__ip_sock_saddr(@entry($sk)), __ip_sock_family(@entry($sk)))
+    daddr = format_ipaddr(__ip_sock_daddr(@entry($sk)), __ip_sock_family(@entry($sk)))
+    sport = __tcp_sock_sport(@entry($sk))
+    dport = __tcp_sock_dport(@entry($sk))
+
+    if (size > 0)
```
Comment (siddontang) [inline systemtap/tools/net_trace.stp]: why check size here?

Comment (LinuxGit) [reply systemtap/tools/net_trace.stp]: Size: Number of bytes received or error code if an error occurred, error code cloud be negative.

Comment (siddontang) [reply systemtap/tools/net_trace.stp]: add a comment here

**Comment by siddontang** [comment]
Rest LGTM

File: systemtap/tools/net_trace.stp (hunk 2 above)
Comment (siddontang) [reply systemtap/tools/net_trace.stp]: what is "cloud be"?

**Pull request merged**